finetune/lora.py

```diff
@@ -226,6 +236,10 @@ def train(
             checkpoint_path = out_dir / f"iter-{iter_num:06d}-ckpt.pth"
             save_lora_checkpoint(fabric, model, checkpoint_path)
 
+    if fabric.device.type == "cuda":
+        fabric.print(f"Memory used: {torch.cuda.max_memory_allocated() / 1e9:.02f} GB")
+
+
```

Comment:
2 + 1 = 3 😄 
Two blank lines are added, one was before, three in total. Needs to be only two between functions.

P.S. That kind of stuff can be easily covered by a style formatter, which I assume you guys don't like 🥲 .
The same goes to imports. 
> from typing import Optional, List, Dict, Tuple, Literal

A, B, C, .... O, L, D, T, L, .... :)

Reply:
We haven't implemented a formatter to keep the repository simpler. I run it manually on the repo from time to time. Just did: https://github.com/Lightning-AI/lit-gpt/commit/866baf228672533e1e7d083b947777b66b840749

Reply:
Then maybe add also [I001](https://beta.ruff.rs/docs/rules/unsorted-imports/) rule? To sort the imports.

Reply:
Should I add pre-commit hooks with a style check (and formatter) @carmocca ? -- separate PR of course

Reply:
William and Luca strongly opposed adding all the required files and directories required to configure them.

@Andrei-Aksionov sure, why not. Feel free to open a pr running that rule on the repo

Reply:
> William and Luca strongly opposed adding all the required files and directories to configure them.

noted